### PR TITLE
#161338201 Harmonise UI and clear input fields on submit

### PR DESCRIPTION
**Pull request by pabishai**
#### What does this PR do?
Harmonizes signup, login, reset password and enter new password modals

#### Description of tasks to be completed
- Harmonize all authentication modals
- Clear input fields on submit

#### How should this be manually tested?
- `Git clone https://github.com/andela/mag-ah-frontend`
- `git checkout bg-clear-state-on-submit-161338201`
- `npm install` to install all the dependancies
- Click on get started
- Click on login
- Click on forgot password
- The UIs should look similar

#### What are the relevant pivotal tracker stories?
[Fix CORS whitelist](https://www.pivotaltracker.com/story/show/161338201)

#### Screenshots
Sign up Form

screen shot 2018-10-22 at 16 53 24

Sign in Form

screen shot 2018-10-22 at 16 53 36

Forgot Password Form

screen shot 2018-10-22 at 16 54 34

Update password form 

screen shot 2018-10-22 at 16 55 43

### CODE REVIEW

**Comment by micahoriaso** [comment]
this looks good to me. Just rebase now.

**Comment by Bileonaire** [comment]
Looks good to me.

File: src/components/auth/ForgotPasswordComp.jsx [hunk 1]
```diff
@@ -3,12 +3,26 @@ import { connect } from "react-redux";
...
 export class ForgotPassword extends Component {
   state = {
     user: { email: "", client_url: window.location.href }
   };
 
+  componentDidMount() {
+    const { dispatch } = this.props;
+    const registerModal = document.querySelector(
+      "#ahResetPasswordModalCloseButton"
+    );
```
Comment (kimotho-njoki) [inline src/components/auth/ForgotPasswordComp.jsx:17]: Empty line please before this if statements

File: src/redux/reducers/ResetPassword.js [hunk 2]
```diff
@@ -19,6 +19,9 @@ export default (state = initialState.resetPassword, action) => {
         error: action.error,
         info: false
       };
+    case RESET_STATE:
+      return initialState;
+
```
Comment (kimotho-njoki) [inline src/redux/reducers/ResetPassword.js]: Why is this empty line here

**Comment by kimotho-njoki** [comment]
Good job people. Please handle the nitpicks then alert me to merge this.

File: src/components/auth/ForgotPasswordComp.jsx (hunk 1 above)
Comment (pabishai) [reply src/components/auth/ForgotPasswordComp.jsx:17]: All done

**Comment by pabishai** [comment]
Handled the spacing

File: src/components/auth/ForgotPasswordComp.jsx (hunk 1 above)
Comment (pabishai) [reply src/components/auth/ForgotPasswordComp.jsx:17]: added the empty line

**Pull request merged**